onyxia-api/src/main/java/fr/insee/onyxia/api/controller/pub/HealthcheckController.java

```diff
@@ -0,0 +1,42 @@
+package fr.insee.onyxia.api.controller.pub;
+
+import fr.insee.onyxia.api.configuration.CatalogWrapper;
+import fr.insee.onyxia.api.configuration.Catalogs;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@Tag(name = "Public")
+@RequestMapping("/public")
+public class HealthcheckController {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(HealthcheckController.class);
+
+    @Autowired private Catalogs catalogs;
```

Comment:
Use constructor based injection, instead of field based. This is described on sonarcloud (so I will not repeate it): https://sonarcloud.io/project/issues?resolved=false&pullRequest=353&id=InseeFrLab_onyxia-api&open=AY1ayCTVgs7rU-KXZaPs&tab=why